source/automem/traits.d

```diff
@@ -43,3 +43,147 @@ template isTheAllocator(Allocator) {
     import std.experimental.allocator: theAllocator;
     enum isTheAllocator = is(Allocator == typeof(theAllocator));
 }
+
+/**
+   Determines if a type is Unique with a supported Allocator.
+ */
+template isUnique(T) {
+    import automem.unique: Unique;
+    import std.typecons: Flag;
+
+    static if (is(T == Unique!(Type, Allocator, supportsGC), 
+                  Type, Allocator, Flag!"supportGC" supportsGC))
+    {
+        static if (isAllocator!Allocator)
+            enum bool isUnique = true;
+        else
+            enum bool isUnique = false;
+    }
+    else
+    {
+        enum bool isUnique = false;
+    }
+}
+
+///
+@("isUnique")
+@safe unittest {
+    import automem.unique: Unique;
+
+    static struct Point {
+        int x;
+        int y;
+    }
+
+    auto u = Unique!Point(2, 3);
+    static assert(isUnique!(typeof(u)));
+
+    auto p = Point(2, 3);
+    static assert(!isUnique!(typeof(p)));
+}
+
+/**
+   The base type of a `Unique` pointer.
+ */
+template UniqueType(T)
+{
+    import automem.unique: Unique;
```

Comment:
Assuming one does `alias UniqueType = Type;` in the definition of `Unique` itself:

```d
static assert(isUnique!T);
alias UniqueType = T.UniqueType;
```

Reply:
That's a good idea. I will try to add that when I have a chance.

Reply:
@atilaneves After thinking on it, I think it would make sense to change the signature of `Unique` so that what is now `Type` is called `UniqueType` or something and then do an alias to `Type` instead. That way code that is generic across `Unique` and `RefCounted` types do not become more complicated. What do you think?

Reply:
@atilaneves See changes.